pkg/i2gw/provider.go

```diff
@@ -32,17 +34,34 @@ import (
 // func at startup.
 var ProviderConstructorByName = map[ProviderName]ProviderConstructor{}
 
+// ProviderConfByName ProviderConstructorByName is a map of ProviderConf by a
+// provider name. Different Provider implementations should add their conf at startup.
+// var ProviderConfByName = map[ProviderName]ProviderConf{}
+
 // ProviderName is a string alias that stores the concrete Provider name.
 type ProviderName string
 
 // ProviderConstructor is a construction function that constructs concrete
 // implementations of the Provider interface.
-type ProviderConstructor func(conf *ProviderConf) Provider
+type ProviderConstructor func() Provider
 
 // ProviderConf contains all the configuration required for every concrete
 // Provider implementation.
 type ProviderConf struct {
```

Comment:
why did we remove the client from `ProviderConf`?  Initially we assume there can be providers that needs a client with different settings. 

I have no strong feelings though, wdyt? 
@mlavacca @levikobi

Reply:
I have no strong feelings either. I'll add again the client and ensure that it is not instantiated when reading from a file